pkg/operator/starter.go

```diff
@@ -194,6 +173,39 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		configClient.ConfigV1(),
 		configInformers.Config().V1().ClusterOperators(),
 		versionRecorder,
+	).WithWorkloadController(
+		"OpenShiftAPIServer",
+		operatorclient.OperatorNamespace,
+		operatorclient.TargetNamespace,
+		os.Getenv("OPERATOR_IMAGE_VERSION"),
+		"openshift",
+		kubeClient,
+		openShiftAPIServerWorkload,
+		configClient.ConfigV1().ClusterOperators(),
+		versionRecorder,
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Namespaces().Informer(),
+		operatorConfigInformers.Operator().V1().OpenShiftAPIServers().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ServiceAccounts().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Services().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Apps().V1().Deployments().Informer(),
+		kubeInformersForNamespaces.InformersFor(metav1.NamespaceSystem).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorclient.GlobalUserSpecifiedConfigNamespace).Core().V1().ConfigMaps().Informer(),
```

Comment:
would have expected lots of this to be hidden in the generic controller.

Reply:
as it is now these reactors will also eventually trigger the custom (provided) `sync` function that is different for `openshift-apiserver` and `oauth-apiserver`